src/core/jsproxy.c

```diff
@@ -1239,15 +1224,90 @@ static PyMethodDef JsProxy_object_values_MethodDef = {
   METH_NOARGS,
 };
 
+EM_JS_NUM(int, get_length_helper, (JsVal val), {
+  // clang-format off
+  let result;
+  if (typeof val.size === "number") {
+    result = val.size;
+  } else if (typeof val.length === "number") {
+    result = val.length;
+  } else {
+    return -2;
+  }
+  if(result < 0){
+    return -3;
+  }
+  if(result > INT_MAX){
+    return -4;
+  }
+  return result;
+  // clang-format on
+});
+
+// Needed to render the length accurately when there is an error
+EM_JS_REF(char*, get_length_string, (JsVal val), {
+  let result;
+  // clang-format off
+  if (typeof val.size === "number") {
+    result = val.size;
+  } else if (typeof val.length === "number") {
+    result = val.length;
+  }
+  // clang-format on
+  return stringToNewUTF8(" " + result.toString())
+})
+
+static int
+get_length(JsVal obj)
+{
+  int result = get_length_helper(obj);
+  if (result >= 0) {
+    return result;
+  }
+  // Something went wrong. Case work:
+  // * -1: Either `val.size` or `val.length` was a getter which managed to raise
+  //    an error. Rude. (Also we don't defend against this in hiwire_has_length)
+  // * -2: Doesn't have a length or size, or they aren't of type "number".
+  //   But `hiwire_has_length` returned true? So it must have changed somehow.
+  // * -3: Length was >= 2^{31}
+  // * -4: Length was negative
```

Comment:
I think the comment of `-3` and `-4` is reversed.

Reply:
This is probably a good place to use `#define`s.